e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/NavigationMenuAO.java

```diff
@@ -15,44 +15,64 @@
  */
 package com.epam.pipeline.autotests.ao;
 
+import org.openqa.selenium.By;
+
 import static com.codeborne.selenide.Condition.exist;
 import static com.codeborne.selenide.Condition.not;
 import static com.codeborne.selenide.Condition.visible;
 import static com.codeborne.selenide.Selectors.*;
 import static com.codeborne.selenide.Selenide.$;
+import static com.epam.pipeline.autotests.utils.Conditions.selectedMenuItem;
+import static com.epam.pipeline.autotests.utils.Utils.click;
 import static com.epam.pipeline.autotests.utils.Utils.sleep;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
 public class NavigationMenuAO {
 
     public PipelinesLibraryAO library() {
-        $(byId("navigation-button-pipelines")).shouldBe(visible).click();
+        final By pipelinesPageSelector = byId("navigation-button-pipelines");
+        click(pipelinesPageSelector);
+        if (!$(pipelinesPageSelector).has(selectedMenuItem)) {
+            click(pipelinesPageSelector);
+        }
         $(byXpath("//*[.//*[text()[contains(.,'Library')]] and contains(@id, 'pipelines-library-content')]"))
                 .waitUntil(visible, 5000);
         return new PipelinesLibraryAO();
     }
 
     public RunsMenuAO runs() {
-        $(byId("navigation-button-runs")).shouldBe(visible).click();
+        final By runsPageSelector = byId("navigation-button-runs");
+        click(runsPageSelector);
+        if (!$(runsPageSelector).has(selectedMenuItem)) {
```

Comment:
I suppose we should not use `has` because it returns immediately. According to the original issue we should wait until the menu element is active rather then trying to click second time. From my point of view usage of simple `should` can be of help here.

Reply:
It is related to all similar use cases in the pull request.

Reply:
@tcibinan Thanks, fixed.